document-store/src/integrationTest/java/org/hypertrace/core/documentstore/DocStoreQueryV1Test.java

```diff
@@ -1977,8 +1982,201 @@ public void testAtomicUpdateDocumentWithoutSelections(final String datastoreName
+          SubDocumentUpdate.builder()
+              .subDocument("props.removed.list")
+              .operator(REMOVE_ALL_FROM_LIST)
+              .subDocumentValue(SubDocumentValue.of(new String[] {"Hello"}))
+              .build();
+
+      final Query query = Query.builder().build();
+      final List<SubDocumentUpdate> updates = List.of(set, unset, add, another_add, append, remove);
+
+      final CloseableIterator<Document> iterator =
+          collection.bulkUpdate(
+              query, updates, UpdateOptions.builder().returnDocumentType(AFTER_UPDATE).build());
+      assertDocsAndSizeEqualWithoutOrder(
+          datastoreName, iterator, "query/update_operator/updated1.json", 9);
+
+      final SubDocumentUpdate set_new =
+          SubDocumentUpdate.builder()
+              .subDocument("props.sales")
+              .operator(SET)
+              .subDocumentValue(SubDocumentValue.of("new_value"))
+              .build();
+      final SubDocumentUpdate unset_new =
+          SubDocumentUpdate.builder()
+              .subDocument("props.new_property.deep.nested")
+              .operator(UNSET)
+              .build();
+      final SubDocumentUpdate add_new =
+          SubDocumentUpdate.builder()
+              .subDocument("props.added.set")
+              .operator(ADD_TO_LIST_IF_ABSENT)
+              .subDocumentValue(SubDocumentValue.of(new Integer[] {3, 1, 1000}))
+              .build();
+      final SubDocumentUpdate append_new =
+          SubDocumentUpdate.builder()
+              .subDocument("props.appended.list")
+              .operator(APPEND_TO_LIST)
+              .subDocumentValue(SubDocumentValue.of(new Integer[] {8, 2}))
+              .build();
+      final SubDocumentUpdate remove_new =
+          SubDocumentUpdate.builder()
+              .subDocument("props.planets")
+              .operator(REMOVE_ALL_FROM_LIST)
+              .subDocumentValue(SubDocumentValue.of(new String[] {"Pluto", "Mars"}))
+              .build();
+      final List<SubDocumentUpdate> new_updates =
+          List.of(set_new, unset_new, add_new, append_new, remove_new);
+
+      final CloseableIterator<Document> iterator_new =
+          collection.bulkUpdate(
+              query, new_updates, UpdateOptions.builder().returnDocumentType(AFTER_UPDATE).build());
+      assertDocsAndSizeEqualWithoutOrder(
+          datastoreName, iterator_new, "query/update_operator/updated2.json", 9);
+    }
+
+    @ParameterizedTest
+    @ArgumentsSource(AllProvider.class)
+    void testRemoveAllOccurrancesFromIntegerList(final String datastoreName) throws IOException {
+      final Collection collection = getCollection(datastoreName, UPDATABLE_COLLECTION_NAME);
+      createCollectionData("query/updatable_collection_data.json", UPDATABLE_COLLECTION_NAME);
+
+      final SubDocumentUpdate unset =
+          SubDocumentUpdate.builder().subDocument("sales").operator(UNSET).build();
+      final SubDocumentUpdate add =
```

Comment:
Is the default operator `SET`? Can we explicitly provide the operator? Also nit: `add` -> `set`

Reply:
> Is the default operator SET?

Yes.

> Can we explicitly provide the operator?

Sure. Will update.